### ISSUE TITLE
extract mesh failed，'list' object has no attribute 'vertices'

### ISSUE THREAD

**Issue by Islaster** [NVlabs/neuralangelo#87]
SOFTWARE:
- Windows 10
- ubuntu-wsl
- Python Anaconda

HARDWARE:
- GeForce 4070 ti
- 208 ram
- SUPERMICRO MOTHERBOARD

![image](https://github.com/NVlabs/neuralangelo/assets/52678788/6cbf416b-b37b-4f4a-8eab-09f257f70d61)

**Comment by mli0603** [comment]
Hi @Islaster 

It looks like no vertices are extracted. 

- Are you on the latest `main` branch?
- If so, this may be related to the checkpoint loading issue of #83. Can you provide your config and checkpoint so I can test it on my end?

**Comment by Islaster** [comment]
just did a pull, what did you mean by config and checkpoint?

**Comment by Islaster** [comment]
CHECKPOINT=logs/example_group/example_name/epoch_00000_iteration_000000010_checkpoint.pt
CONFIG=logs/example_group/example_name/config.yaml

**Comment by Islaster** [comment]
the issue you sent to is for training im having problems with the mesh extraction

**Comment by mli0603** [comment]
Hi @Islaster 

- Yes you found the right files. Could you share the files so I can download and test them?
- The issue is related to loading the checkpoint, which affects mesh extraction as well.

**Comment by Islaster** [comment]
it's not letting me copy it into the comment so I'm going to send google drive links

**Comment by Islaster** [comment]
config file:
https://drive.google.com/file/d/1-CJBpAIHi-4xlW-SJ1KvkQOaCE5c34H2/view?usp=sharing
checkpoint:
https://drive.google.com/file/d/1PReVRmim17iAkiXysjTzUPjJiKoOqvp3/view?usp=sharing

**Comment by mli0603** [comment]
Thanks @Islaster 

I can confirm that this is a bug potentially related to #83 where your checkpoint saved is corrupted and all the SDF values predicted are `NaN`. Therefore, no meshes are found.

**Comment by Islaster** [comment]
so how do i fix it?

**Comment by mli0603** [comment]
Can you provide more information on the following?

- I see in your config file that your `save_iter` is 20000. But your checkpoint is saved at iteration 10. How did you save this checkpoint?
- Can you provide the set of commands you used for training?
- Can you provide your pytorch version?

**Comment by Islaster** [comment]
EXPERIMENT=template
GROUP=example_group
NAME=example_name
CONFIG=projects/neuralangelo/configs/custom/${EXPERIMENT}.yaml
GPUS=1  
torchrun --nproc_per_node=${GPUS} train.py \
    --logdir=logs/${GROUP}/${NAME} \
    --config=${CONFIG} \
    --show_pbar

**Comment by Islaster** [comment]
pytorch version 2.0.0
and I changed the max iter in base.yaml

**Comment by Islaster** [comment]
I can send you my whole git clone if that would help clear things up

**Comment by mli0603** [comment]
Hi @Islaster 

I was able to run your entire pipeline with torch version 1.12.1. Here is the mesh extracted at iteration 10 using the Lego dataset.

![image](https://github.com/NVlabs/neuralangelo/assets/13513172/c41ab0b7-7ac6-4dc4-bebe-0de39fad2c80)

**Comment by Islaster** [comment]
will do

**Comment by Islaster** [comment]
what version of anaconda do you use?

**Comment by mli0603** [comment]
I am using conda 4.10.1. We have also pushed a latest update of the code that could potentially fix the issue. Can you pull the latest version and try it out on your end?

**Comment by Islaster** [comment]
I'm getting conflict errors when downloading Pytorch 1.12.1 
could it have something to do with the anaconda envs files for neuralangelo

**Comment by lie12huo** [comment]
I had the same problem, is it because of too few iterations for training? I changed the configuration items save_epoch, max_epoch and max_iter in config_base.yaml to 3. Are there any minimum requirements for these values?

**Comment by mli0603** [comment]
Hi @lie12huo 

- No, it is not because training is too few iterations. It is likely the saved checkpoint has all 0s for the model weights. Can you check?
- Are you running the latest code? If not, we recently pushed a commit that fixes this issue, which potentially is torch-version related.

**Comment by lie12huo** [comment]
I used the latest code for training and model extraction, but the data preprocessing did not use the latest version of the code. Does this affect the result? How do I check if there is a problem with the saved checkpoint? By the way, I am currently using the test data toy_example.mov.

**Comment by lie12huo** [comment]
@mli0603 
I reprocessed the data for training, generated new checkpoints, and exported the mesh, but the result was poor and there was no texture.
![image](https://github.com/NVlabs/neuralangelo/assets/52315796/842b8000-28dc-4b1e-8bed-edde144e23e6)
Is this because 50 epochs are too few? Or is it related to my adjustment of parameters dict_size: 21, dim: 4?

**Comment by mli0603** [comment]
Hi @lie12huo 

It's likely undertrained. You can also take a look at the [colab](https://colab.research.google.com/drive/13u8DX9BNzQwiyPPCB7_4DbSxiQ5-_nGF) for how to obtain good results.
